common/src/main/java/gov/cms/ab2d/common/httpclient/APIClient.java

```diff
@@ -0,0 +1,173 @@
+                .build();
+
+        this.ab2dApiUrl = ab2dApiUrl;
+        this.oktaUrl = oktaUrl;
+        authEncoded = Base64.getEncoder().encodeToString((oktaClientId + ":" + oktaPassword).getBytes());
+
+        generateToken();
+    }
+
+    // Could make public later if client needs to regenerate
+    private void generateToken() throws JSONException, IOException, InterruptedException {
+        var jwtRequestParms = new HashMap<>() {{
+            put("grant_type", "client_credentials");
+            put("scope", "clientCreds");
+        }};
+
+        HttpRequest jwtRequest = HttpRequest.newBuilder()
+                .uri(URI.create(oktaUrl))
+                .timeout(Duration.ofSeconds(30))
+                .header("Content-Type", "application/x-www-form-urlencoded")
+                .header("Accept", "application/json")
+                .header("Authorization", "Basic " + authEncoded)
+                .POST(buildFormDataFromMap(jwtRequestParms))
+                .build();
+
+        HttpResponse<String> jwtResponse = httpClient.send(jwtRequest, HttpResponse.BodyHandlers.ofString());
+        String responseJwtString = jwtResponse.body();
+
+        log.debug("Received JWT response {}", responseJwtString);
+
+        JSONObject responseJsonObject = new JSONObject(responseJwtString);
+        jwtStr = responseJsonObject.getString("access_token");
+    }
+
+    public HttpResponse<String> exportRequest() throws IOException, InterruptedException {
+        HttpRequest exportRequest = HttpRequest.newBuilder()
+                .uri(URI.create(ab2dApiUrl + PATIENT_EXPORT_PATH))
+                .timeout(Duration.ofSeconds(defaultTimeout))
+                .header("Content-Type", "application/json")
+                .header("Authorization", "Bearer " + jwtStr)
+                .GET()
+                .build();
+
+        return httpClient.send(exportRequest, HttpResponse.BodyHandlers.ofString());
+    }
+
+    public HttpResponse<String> exportByContractRequest(String contractNumber) throws IOException, InterruptedException {
+        HttpRequest exportRequest = buildExportByContractRequest(contractNumber);
+
+        return httpClient.send(exportRequest, HttpResponse.BodyHandlers.ofString());
+    }
+
+    public HttpRequest buildExportByContractRequest(String contractNumber) {
+        return HttpRequest.newBuilder()
+                .uri(URI.create(ab2dApiUrl + "Group/" + contractNumber + "/$export"))
+                .timeout(Duration.ofSeconds(defaultTimeout))
+                .header("Content-Type", "application/json")
+                .header("Authorization", "Bearer " + jwtStr)
+                .GET()
+                .build();
+    }
+
+    public CompletableFuture<HttpResponse<String>> exportByContractRequestAsync(String contractNumber) {
+        HttpRequest exportRequest = buildExportByContractRequest(contractNumber);
+
+        return httpClient.sendAsync(exportRequest, HttpResponse.BodyHandlers.ofString());
+    }
```

Comment:
Are we using this?

Reply:
I was using that in some of the e2e testing before but I can delete now.